metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java

```diff
@@ -0,0 +1,258 @@
+            hiveConnectorDatabaseService, hiveMetacatConverters, Boolean.parseBoolean(
+            metacatConnectorConfig.getConfiguration()
+                .getOrDefault(HiveConfigConstants.ALLOW_RENAME_TABLE, "false")));
+    }
+
+    /**
+     * create hive connector fast table service.
+     *
+     * @param metacatHiveClient            metacat hive client
+     * @param hiveMetacatConverters        hive metacat converters
+     * @param hiveConnectorDatabaseService hive database service
+     * @param metacatConnectorConfig       server context
+     * @param dataSource                   data source
+     * @return HiveConnectorFastTableService
+     */
+    @Bean
+    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "true")
+    public HiveConnectorFastTableService fastHiveTableService(
+        final IMetacatHiveClient metacatHiveClient,
+        final HiveConnectorInfoConverter hiveMetacatConverters,
+        final HiveConnectorDatabaseService hiveConnectorDatabaseService,
+        final MetacatConnectorConfig metacatConnectorConfig,
+        @Qualifier("hiveDataSource") final DataSource dataSource
+
+    ) {
+        return new HiveConnectorFastTableService(
+            metacatConnectorConfig.getCatalogName(), metacatHiveClient,
+            hiveConnectorDatabaseService, hiveMetacatConverters,
+            Boolean.parseBoolean(
+                metacatConnectorConfig.getConfiguration()
+                    .getOrDefault(HiveConfigConstants.ALLOW_RENAME_TABLE, "false")),
+            metacatConnectorConfig,
+            dataSource);
+    }
+
+    /**
+     * create local hive client.
+     *
+     * @param metacatConnectorConfig connector config context
+     * @return IMetacatHiveClient
+     * @throws Exception exception
+     */
+
+    @Bean
+    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "true")
+    public IMetacatHiveClient createLocalClient(
+        final MetacatConnectorConfig metacatConnectorConfig) throws Exception {
+        try {
+            final HiveConf conf = getDefaultConf();
+            metacatConnectorConfig.getConfiguration().forEach(conf::set);
+            DataSourceManager.get().load(metacatConnectorConfig.getCatalogName(),
+                metacatConnectorConfig.getConfiguration());
+            return new EmbeddedHiveClient(metacatConnectorConfig.getCatalogName(),
+                HMSHandlerProxy.getProxy(conf), metacatConnectorConfig.getRegistry());
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                String.format("Failed creating the hive metastore client for catalog: %s",
+                    metacatConnectorConfig.getCatalogName()), e);
+        }
+    }
+
+    /**
+     * hive DataSource.
+     *
+     * @param metacatConnectorConfig connector config.
+     * @return data source
+     */
+    @Bean
+    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "true")
```

Comment:
You have this conditional a lot. I think what you should do in this case is create a whole new `Config` class which the whole class is conditional on this property being true and wire all the beans you need in there. Leave this config class as the default. That will be much cleaner and easier to understand what's going on.